calico_containers/integrations/kubernetes/calico_kubernetes.py

```diff
@@ -336,6 +358,97 @@ def _apply_tags(self, profile_name, pod):
                 print('Could not create tag %s.\n%s' % (tag, e))
         print('Finished applying tags.')
 
+    def _get_annotations(self, pod):
+        """
+        Return annotations given Pod Object
+        Returns None if no annotation exists
+        """
+        print('Getting Annotations for pod %s' % pod)
+        try:
+            annotations = pod['metadata']['annotations']
+            print('Annotations for Pod %s\n%s' % (pod, annotations))
+            return annotations
+        except KeyError:
+            print('No Annotations found in pod %s' % pod)
+            return None
+
+    def _get_namespace(self, pod):
+        """
+        Return Namespace given Pod Object
+        Returns None if no namespace exists
+        """
+        try:
+            namespace = pod['metadata']['namespace']
+            print("Pod %s in Namespace %s" % (pod, namespace))
+            return namespace
+        except KeyError:
+            print('No namespace found in pod %s' % pod)
+            return None
+
+    def _label_to_tag(self, label_key, label_value, namespace):
+        """
+        Labels are key-value pairs, tags are single strings. This function handles that translation
+        1) concatenate key and value with '='
+        2) prepend a pod's namespace followed by '/' if available
+        3) replace special characters with urllib-style escape sequence
+        :param label_key: key to label
+        :param label_value: value to given key for a label
+        :param namespace: Namespace string, input None if not available
+        :param types: (self, string, string, string)
+        :return single string tag
+        :rtype string
+        """
+        escape_seq = '_'
+
+        tag = '%s=%s' % (label_key, label_value)
+        tag = (namespace + '/' + tag) if namespace else tag
+        tag = quote(tag, safe='')
+        tag = tag.replace('%', escape_seq)
+
+        return tag
+
+    def _translate_rule(self, rule, namespace):
+        """
+        Given a JSON dict rule from Kubernetes Annotations, 
+        output a JSON dict that is calicoctl profile compliant
+        :param rule: dict of keys relating to policy rules
+        :param namespace: string indicating namespace for pod, None if not available
+        :return: translated dict of keys relating to policy rules for Calico profile
+        :rtype: JSON dict
+        """
+        calico_rule = rule
```

Comment:
This doesn't actually make a copy of rule - they two variables will use the same underlying dictionary.